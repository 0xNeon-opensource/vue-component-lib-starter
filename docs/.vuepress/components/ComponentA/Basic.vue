<template>
  <ComponentA msg="Hello"/>
</template>
