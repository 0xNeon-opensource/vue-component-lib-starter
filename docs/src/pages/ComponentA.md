# Component A

The description for Component A goes here.

### Example

```html
<template>
  <component-a :msg="msg"/>
</template>
<script>
import { ComponentA } from 'mylib';

export default {
  data() {
    return {
      msg: 'Greeting'
    }
  },
  components: {
    ComponentA
  }
}
</script>
<!-- component-demo.vue -->
````

### Props

Name                 | Type       | Default      | Required | Description
----------------     | ---------- | ------------ | -------- | -----------------------
`msg`                | String     | Hello World! |          | A message for Component A
