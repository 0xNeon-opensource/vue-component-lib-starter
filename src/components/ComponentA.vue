<template>
  <div>
    <p>Component A has 'msg' prop: <span class="message">{{ msg }}</span></p>
  </div>
</template>

<script>
export default {
  props: {
     msg: {
       type: String,
       required: 'Hello World!'
     }
  }
}
</script>

<style lang="scss" scoped>
.message {
  color: #42b983;
}
</style>