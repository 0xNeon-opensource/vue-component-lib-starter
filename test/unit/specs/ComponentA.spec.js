import Vue from 'vue';
import { mount } from 'vue-test-utils';
import ComponentA from '@/components/ComponentA';

const createCmp = propsData => mount(ComponentA, { propsData });

describe('ComponentA.vue', () => {
  let cmp;

  it('has a msg property', () => {
    cmp = createCmp({ msg: 'hey' });
    expect(cmp.props('msg', 'hey')).toBeTruthy();
  });

  it('has the expected html structure', () => {
    cmp = createCmp({ msg: 'Greeting' });
    expect(cmp.element).toMatchSnapshot();
  });
})