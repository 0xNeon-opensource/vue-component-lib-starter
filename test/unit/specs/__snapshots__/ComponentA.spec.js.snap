// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentA.vue has the expected html structure 1`] = `
<div
  class="component-a"
>
  <p>
    Component A has 'msg' prop: 
    <span
      class="message"
    >
      Greeting
    </span>
  </p>
</div>
`;
